### PR TITLE
HDF5 interface (WIP)

### PR DESCRIPTION
**Pull request by ghost**
I am leaving it in the beta folder because I still need to test it. Not to mention the improvements made since the root fork of this branch.

Resolves onalant/privileged_residues#4. I hope.

### CODE REVIEW

File: beta/process.py [hunk 1]
```diff
@@ -23,90 +24,122 @@
...
+        minmov = pyrosetta.rosetta.protocols.minimization_packing.MinMover(mmap, sfx, "dfpmin_armijo_nonmonotone", 0.01, False)
+
+        for hit in hits:
+            proto_pose = p.clone()
+            proto_pose.append_pose_by_jump(hit, len(proto_pose.residues))
+
+            sfx(proto_pose)
+
+            fa_rep = proto_pose.energies().total_energies()[scoring.fa_rep]
+            fa_elec = proto_pose.energies().total_energies()[scoring.fa_elec]
```
Comment (ghost) [inline beta/process.py]: Is this too strict?

Comment (weitzner) [reply beta/process.py]: Possibly, but 35 kcal is a pretty big number. I suspect that testing on some real-world examples will be the best way to answer that question.

**Comment by weitzner** [comment]
Hmm do we want to include the notes in the revision history? Is there just the two?

**Comment by ghost** [comment]
Probably not. I just removed them for the time being.

**Comment by weitzner** [comment]
Cool cool cool. Could you make the `beta/basic_matcher` more closely mirror the root-level `basic_matcher`?

**Comment by ghost** [comment]
Will do.

Just a side note: I am probably going to strip out the type annotations, because they are useless and just add visual clutter.

________________________________
Gönderen: Brian Weitzner <notifications@github.com>
Gönderildi: Wednesday, June 20, 2018 7:50:38 PM
Kime: weitzner/privileged_residues
Bilgi: Tarik Onalan; Author
Konu: Re: [weitzner/privileged_residues] New Interface (with HDF5 🎉🎊) (#6)


Cool cool cool. Could you make the beta/basic_matcher more closely mirror the root-level basic_matcher?

**Comment by weitzner** [comment]
Yeah, adding google-formatted docstrings is preferable.

http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: beta/table.py [hunk 2]
```diff
@@ -2,35 +2,29 @@
...
-from lru import LRU
-from typing import Callable, Iterable, Mapping, Tuple, Union
-
 # NOTE(onalant): <C-g> will show current file in nvi!
 
-class GenericTable(Mapping[np.uint64, np.ndarray]):
+class GenericTable:
 
-    def __init__(self, dbpath: str, lru: int = 0) -> None:
+    def __init__(self, dbpath) -> None:
```
Comment (weitzner) [inline beta/table.py]: still has return type!

Comment (weitzner) [reply beta/table.py]: That felt a bit like finding Waldo.

Comment (ghost) [reply beta/table.py]: Fixed in 8ab984d.

**Comment by ghost** [comment]
By `beta/basic_matcher.py`, do you mean `beta/demo.py`?'

I need to finish retooling the bidentate selection. Working on it now.

**Comment by onalant** [comment]
I need to finish retooling the bidentate selection. Working on it now.

**Comment by weitzner** [comment]
I meant `beta/test/basic_matcher.py`, but if the intention is to remove that and update the root-level `basic_matcher.py`, that's fine too.

**Comment by weitzner** [comment]
Also, perhaps it would be wiser to move the files into their intended home but in a branch so you always have a working version. In general, a branch named "beta" is superior to a directory named the same way. Keep up the good work!

**Pull request closed without merging**